x-pack/legacy/plugins/uptime/server/lib/adapters/monitor_states/elasticsearch_monitor_states_adapter.ts

```diff
@@ -58,42 +53,36 @@ export class ElasticsearchMonitorStatesAdapter implements UMMonitorStatesAdapter
       nextPagePagination: jsonifyPagination(page.nextPagePagination),
       prevPagePagination: jsonifyPagination(page.prevPagePagination),
     };
-  }
+  },
 
-  public async getSnapshotCount(
-    request: any,
-    dateRangeStart: string,
-    dateRangeEnd: string,
-    filters?: string,
-    statusFilter?: string
-  ): Promise<Snapshot> {
+  getSnapshotCount: async ({ callES, dateRangeStart, dateRangeEnd, filters, statusFilter }) => {
     const context: QueryContext = {
-      database: this.database,
-      request,
+      count: query => callES('count', query),
+      search: query => callES('search', query),
       dateRangeStart,
       dateRangeEnd,
       pagination: CONTEXT_DEFAULTS.CURSOR_PAGINATION,
       filterClause: filters && filters !== '' ? JSON.parse(filters) : null,
       size: CONTEXT_DEFAULTS.MAX_MONITORS_FOR_SNAPSHOT_COUNT,
-      statusFilter,
+      statusFilter: statusFilter || undefined,
```

Comment:
i wonder why are you explicitly assigning it as undefined, if it's already undefined, i mean that's how short-hand objection notation works.

Reply:
Reason for this: QueryContext requires `statusFilter` to be `string | undefined`, but the typing for `statusFilter` from API requests historically can be `string | null | undefined`. I didn't want to break any code that might be explicitly testing the value for `undefined` so I overwrite any falsy value with `undefined`. We can clean this up by modifying the type provided by the request handler, which I will do.

Reply:
Fixed this in https://github.com/elastic/kibana/pull/51403/commits/c0d518714978ce6abea8887de953709f378b5691.